### PR TITLE
feat(send-notification) : Select and send notification

### PR DESCRIPTION
**Pull request by anoj-pushowl**
PR Consist of feature
1. Create notification
2. Show list of notifications
3. Select and send notification

Screenshots

Screenshot 2023-02-09 at 7 50 58 PM

Screenshot 2023-02-09 at 7 51 35 PM

### CODE REVIEW

File: push_notification_ui/static/js/main.js [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+    createNotification(notification_form)
 });
 
+reload_notification.addEventListener('click', function(event) {
+    event.preventDefault()
+    getNotificationItems(notification_items)
+})
+
+send_notification_btn.addEventListener('click', function(event) {
+    event.preventDefault()
```
Comment (GeekyShacklebolt) [inline push_notification_ui/static/js/main.js]: Why do we want to `preventDefault` of `click` events?

File: push_notification_ui/static/js/notifications.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        .then(data => {
+            data.forEach(notification => render(notification, notification_items))
+        });
+}
+
+function render(notification, notification_items) {
+    const option = document.createElement('option')
+    option.id = notification.id
+    option.appendChild(document.createTextNode(`${notification.title}`))
+    notification_items.appendChild(option)
```
Comment (GeekyShacklebolt) [inline push_notification_ui/static/js/notifications.js:15]: Nice! Does it handle the case if we click `reload_notification` button more than once? We would not want to fill `option` with duplicated notifications upon refreshing the list.

File: push_notification_ui/static/js/requests.js [hunk 3]
```diff
@@ -24,12 +24,60 @@ export function registerSubscriber(subscriptionData){
...
+
+export function sendNotification(notification_id) {
+    let url = getBaseUrl() + '/api/v1/notification/send/?id='+notification_id
+    let headers = {
+        'Content-type': 'application/json; charset=UTF-8',
+    }
+    return makeHttpRequest('POST', url, headers)
+        .then((responseJson) => {
+            if (responseJson['success']){
+                alert("Notification send successfully")
```
Comment (GeekyShacklebolt) [inline push_notification_ui/static/js/requests.js]: Thought:
At this point we can't say that notification is sent/received successfully. Or can we?
According to me we can only say that "Notification is scheduled to be sent successfully."

File: push_notification_ui/static/js/requests.js [hunk 4]
```diff
@@ -24,12 +24,60 @@ export function registerSubscriber(subscriptionData){
...
+            } else {
+                alert("Failed to send notification")
+            }
+            return responseJson
+        }).catch(error => console.error('Error: ', error))
 }
```
Comment (GeekyShacklebolt) [inline push_notification_ui/static/js/requests.js:83]: What is the convention in FE regarding this newline at the end of file? In BE we follow that all files should have one blank line at the end of each file.

**Review by GeekyShacklebolt — COMMENTED** [review]
Looking Good! 🌮  Left my questions/thoughts inline.

File: push_notification_ui/static/js/notifications.js (hunk 2 above)
Comment (anoj-pushowl) [reply push_notification_ui/static/js/notifications.js:15]: @GeekyShacklebolt  It's working fine, No duplicate list on clicking refresh button multiple times

File: push_notification_ui/static/js/main.js (hunk 1 above)
Comment (anoj-pushowl) [reply push_notification_ui/static/js/main.js]: Noted, removing default action on click events

**Pull request merged**